pkg/ccl/sqlccl/restore.go

```diff
@@ -779,7 +779,7 @@ func restore(
 		select {
 		case importsSem <- struct{}{}:
 		case <-gCtx.Done():
-			return failed, gCtx.Err()
+			return failed, errors.Wrapf(g.Wait(), "importing %d ranges", len(importSpans))
```

Comment:
That one is inside a g.Go callback and so is correct. But yeah, it's very hard to get this right. Dan filed a note to audit this throughout Bulk I/O. Since you know more about linters: how hard would it be to write a linter to check for this?